dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/analytics/AnalyticsSettings.java

```diff
@@ -60,4 +70,66 @@ public String getTableType() {
 
     return EMPTY;
   }
+
+  private boolean isCitusDisabledByConfig() {
+    return dhisConfigurationProvider.isEnabled(CITUS_EXTENSION_DISABLED);
+  }
+
+  public synchronized boolean isCitusEnabled(JdbcTemplate jdbcTemplate) {
+    if (Objects.isNull(citusEnabled)) {
+      if (isCitusDisabledByConfig()) {
+        log.info("Citus extension is disabled in dhis.conf");
+        citusEnabled = false;
+      } else {
+        citusEnabled = isCitusExtensionInstalledAndEnabled(jdbcTemplate);
+      }
+    }
+    return citusEnabled;
+  }
+
+  private boolean isCitusExtensionInstalledAndEnabled(JdbcTemplate jdbcTemplate) {
+    List<PgExtension> installedExtensions = getPostgresExtensions(jdbcTemplate);
+
+    logFoundExtensions(installedExtensions);
+
+    return installedExtensions.stream().anyMatch(this::isCitusEnabled);
+  }
+
+  private void logFoundExtensions(List<PgExtension> installedExtensions) {
+    if (!installedExtensions.isEmpty()) {
+      log.info(
+          "Found citus extensions in the database: {}",
+          StringUtils.join(installedExtensions, ", "));
+      return;
+    }
+    log.info(
+        "Citus extension software is not installed in the database. You need to install it first, depending on the OS: https://docs.citusdata.com/en/latest/installation/multi_node.html");
+  }
+
+  private List<PgExtension> getPostgresExtensions(JdbcTemplate jdbcTemplate) {
+    return jdbcTemplate.query(
+        "select name, installed_version "
+            + "from pg_available_extensions "
+            + "where lower(name) like 'citus%'",
+        new DataClassRowMapper<>(PgExtension.class));
+  }
+
+  private boolean isCitusEnabled(PgExtension e) {
+    boolean isCitusExtensionEnabled =
```

Comment:
Similar to the question above: should we target only the DHIS database to check if the extension is installed?
Also, can the extension be installed, but disabled at PG level? Is it possible?